### PR TITLE
Documentation update (PPTT-2056)

### PR DESCRIPTION
**Pull request by wronan**
Preview:
https://developers.piwik.pro/en/pptt-2056_documentation/tracker/tracker_debugger_api.html

### CODE REVIEW

**Comment by RomanSek** [comment]
Looks good, but response example is using `"string"` placeholders. Usually we try to give there values that could be received in response.
2021-04-26T095806

**Comment by JacekClearcode** [comment]
Also Master is the target of the PR? :confused:

**Comment by wronan** [comment]
Regarding `master` being the target: sorry, I forgot to ask which branch should be the target and just automatically created the PR.

So, which one should be the target? 15.4? 
If yes, then what is the purpose of the master branch here?

**Comment by wronan** [comment]
@RomanSek content type is correct.
The `string` in the example response value is generated automatically most likely for optional parameters. But I'll look into that once more, maybe I can do something about it.

**Comment by JacekClearcode** [comment]
> Regarding `master` being the target: sorry, I forgot to ask which branch should be the target and just automatically created the PR.
> 
> So, which one should be the target? 15.4?
> If yes, then what is the purpose of the master branch here?

Master = latest. 
About which one, 15.4 for now? I'll merge it to correct one later.

**Comment by RomanSek** [comment]
@wronan Yes, all PR changes should be merged to version where they were introduced in. If we plan to add this API in 15.4 then it's correct now (feature freeze for that version started last Friday).

**Comment by JacekClearcode** [comment]
And if you want to add it to later version, such as 15.5 I also created branch for that.

**Comment by azarecki** [comment]
We want it to version `16.0`. Please add such branch

File: tracker/tracker_object.rst [hunk 1]
```diff
@@ -451,6 +451,12 @@ Cookies that are used by analytics are first party cookies.
 
     The function that will disable all first party cookies. Existing ones will be deleted in the next page view.
 
+.. function:: enableCookies()
```
Comment (azarecki) [inline tracker/tracker_object.rst:454]: Please add it in a separate PR

File: tracker/tracker_debugger_api.yaml [hunk 2]
```diff
@@ -0,0 +1,876 @@
+openapi: 3.0.2
+info:
+  title: Processor Storage API
```
Comment (azarecki) [inline tracker/tracker_debugger_api.yaml:3]: Please change to Tracker API

**Review by azarecki — COMMENTED** [review]
Please apply fixes from https://github.com/PiwikPRO/PPMS-ApiDocs/pull/170/files

**Comment by azarecki** [comment]
Screenshot - 2021-04-26T135610 719
@JacekClearcode @RomanSek can we do something about it?

File: tracker/tracker_object.rst (hunk 1 above)
Comment (wronan) [reply tracker/tracker_object.rst:454]: Hmm... i popped up when I changed the target branch...

Comment (RomanSek) [reply tracker/tracker_object.rst]: Rebase fixed it.

**Comment by RomanSek** [comment]
> Screenshot - 2021-04-26T135610 719
> @JacekClearcode @RomanSek can we do something about it?

I don't thin so. Long names will be a problem with limited space here. @JacekClearcode looked into fixing formatting of this before, maybe he will have a better idea.

File: tracker/tracker_debugger_api.yaml (hunk 2 above)
Comment (adaqus) [reply tracker/tracker_debugger_api.yaml:3]: Using `Tracker API` would introduce another term after `Processor Storage API` and `Tracker debugger API`. It is better to user `Tracker debugger API` everywhere in this doc.

File: tracker/tracker_debugger_api.yaml [hunk 3]
```diff
@@ -0,0 +1,876 @@
...
+            default: 30
+            example: 90
+          in: query
+          required: false
+      responses:
+        '200':
+          content:
+            application/x-ndjson:
+              schema:
+                type: array
```
Comment (adaqus) [inline tracker/tracker_debugger_api.yaml:72]: NDJSON is not array, this is what differentiates it from array of JSON objects.

Comment (adaqus) [reply tracker/tracker_debugger_api.yaml:72]: As I can see we describe this type as an array also in our specification in PSA, I hope this won't cause any misunderstanding.

File: tracker/tracker_debugger_api.yaml [hunk 4]
```diff
@@ -0,0 +1,876 @@
...
+        - plugin_cookie
+        - plugin_director
+      type: object
+      properties:
+        session_id:
+          description: ID of session - random hash
+          type: integer
+        server_time:
+          format: date-time
+          description: 'Time of session cration in UTC in YYYY-MM-DDTHH:MM:SS format in UTC'
```
Comment (adaqus) [inline tracker/tracker_debugger_api.yaml:444]:
```suggestion
          description: 'Time of session creation in YYYY-MM-DDTHH:MM:SS format in UTC'
```

File: tracker/tracker_debugger_api.yaml [hunk 5]
```diff
@@ -0,0 +1,876 @@
...
+          description: Name of PPAS organization tracked website belongs to
+          type: string
+        cookie_id:
+          description: Cookie ID
+          type: integer
+        user_id:
+          description: User ID
+          type: string
+        visitor_id:
+          description: 'Visitor identifier, defaults to `cookie_id` value. When `user_id` is available, this field is set to first 16 characters of `user_id` SH1 hash'
```
Comment (adaqus) [inline tracker/tracker_debugger_api.yaml:467]:
```suggestion
          description: 'Visitor identifier, defaults to `cookie_id` value. When `user_id` is available, this field is set to first 16 characters of `user_id` SHA1 hash'
```

File: tracker/tracker_debugger_api.yaml [hunk 6]
```diff
@@ -0,0 +1,876 @@
...
+        session_ecommerce_status:
+          description: Was there a commerce conversion during a session?
+          enum:
+            - '0'
+            - '1'
+            - '2'
+            - '3'
+          type: string
+        source:
+          description: Source (?)
```
Comment (adaqus) [inline tracker/tracker_debugger_api.yaml:567]:
```suggestion
          description: Referrer source
```

File: tracker/tracker_debugger_api.yaml [hunk 7]
```diff
@@ -0,0 +1,876 @@
...
+          type: string
+        referrer_type:
+          description: Referrer type
+          enum:
+            - '1'
+            - '2'
+            - '3'
+            - '6'
```
Comment (adaqus) [inline tracker/tracker_debugger_api.yaml:575]: Please add '7'.

File: tracker/tracker_debugger_api.yaml [hunk 8]
```diff
@@ -0,0 +1,1234 @@
...
+                type: string
+                example: 404 page not found
+          description: Unauthorized request (eg. `Authorization` header is missing or contains incorrect JWT value)
+        '500':
+          content:
+            atext/plain:
+              schema:
+                type: string
+                example: Internal error occurred. Check logs
+          description: Internal application error response (some service encountered unexpected error and died in pain)
```
Comment (azarecki) [inline tracker/tracker_debugger_api.yaml:108]: Please change the description

**Review by azarecki — APPROVED** [review]

**Comment by RomanSek** [comment]
Mismatched type of response between response example and definition:
2021-04-28T202331
2021-04-28T202347

I don't know how to define correctly this "application/x-ndjson" format. Maybe just give single object in both? Alternatively we could use standard JSON as default example and advise that ndjson is available via [Accept](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept) header.

**Review by RomanSek — APPROVED** [review]

**Review by adaqus — APPROVED** [review]

**Pull request merged**